### DISCUSSION TITLE
Nested dynamic routes?

### DISCUSSION THREAD

**Discussion by craigsmitham** [TanStack/router#2850] [Q&A] [answered]
How would I set up file-based routing with tanstack router so that I could have these three different routes:
* /{year} routes to YearViewComponent
* /{year}/{month} routes to MonthViewComponent
* /{year}/{month}/{day} routes to DayViewComponent

I've tried to accomplish the above by defining three file-based routes:
* $year.$month.$day.tsx
* $year.$month.tsx
* $year.tsx

However, the $year.tsx component is picking up all routes, instead of allowing $year.$month to handle /2025/12 for example.

**Comment by SeanCassiere** [comment, marked as answer]
I think you are looking to set up the necessary `index` routes within the each level.

```
$year.tsx - layout content / or even just empty component
$year.index.tsx - Actual content to be shown when vistsing http://localhost:3000/2024
```

**Reply by craigsmitham** [reply]
Thank you @SeanCassiere !
